indungu/yummy-recipes
# CRUD Features for Recipes and Categories, App staging

### What this PR is for,

The back-end functionality (CRUD) having been well integrated with the front end as proposed/designed [here](https://indungu.github.io/yummy-recipes) warrants that a review of the changes proposed before they are merged to `master`

### Where to start your review

First you can interact with the UI static design files deployed via `gh-pages` on [this link here](https://indungu.github.io/yummy-recipes).
Be sure to note all the missing functionality.
There after, interact with the staged app as deployed on Heroku via [this link here](https://yummy-recipes-indungu.herokuapp.com)

### What to review

The scope of your review should not be limited to the areas I will highlight below, feel free to test every aspect of the staged app especially. All feedback is more than welcome.

1. Test/Review user signup and login features
2. Test/ Review all the CRUD features for the Categories and Recipes

### How to test

For an insight on the test coverage for this app, click the `coverage` badge on either the `master` or the `dev` branch. 
As for how to test the code locally for this PR please refer to the `install` section as documented on the [README.md](https://github.com/indungu/yummy-recipes/tree/dev#yummy-recipes).

You may read the source files as you see fit. To run the unit tests just use this script:
```bash
$ pytest --cov=app tests/ #run tests with coverage reporting
```

### Notes
Please be sure to leave a comment - wherever possible. Thank you.
Apologies for the size of the review files, This PR includes commits for missed unit tests and all the CRUD features in bits.
About 2-3 more commits are projected to be made during the course of this PR, this shall be for minor bug fixes though.
_Your review is much appreciated_
